sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/RetryOptions.java

```diff
@@ -0,0 +1,184 @@
+     *
+     * @param tryTimeout The maximum duration to wait for completion.
+     * @return The updated {@link RetryOptions} object.
+     */
+    public RetryOptions tryTimeout(Duration tryTimeout) {
+        this.tryTimeout = tryTimeout;
+        return this;
+    }
+
+    /**
+     * Gets the approach to use for calculating retry delays.
+     *
+     * @return The approach to use for calculating retry delays.
+     */
+    public RetryMode retryMode() {
+        return retryMode;
+    }
+
+    /**
+     * The maximum number of retry attempts before considering the associated operation to have failed.
+     *
+     * @return The maximum number of retry attempts before considering the associated operation to have failed.
+     */
+    public int maxRetries() {
+        return maxRetries;
+    }
+
+    /**
+     * Gets the delay between retry attempts for a fixed approach or the delay on which to base calculations for a
+     * backoff-approach.
+     *
+     * @return The delay between retry attempts.
+     */
+    public Duration delay() {
+        return delay;
+    }
+
+    /**
+     * Gets the maximum permissible delay between retry attempts.
+     *
+     * @return The maximum permissible delay between retry attempts.
+     */
+    public Duration maxDelay() {
+        return maxDelay;
+    }
+
+    /**
+     * Gets the maximum duration to wait for completion of a single attempt, whether the initial attempt or a retry.
+     *
+     * @return The maximum duration to wait for completion of a single attempt, whether the initial attempt or a retry.
+     */
+    public Duration tryTimeout() {
+        return tryTimeout;
+    }
+
+    /**
+     * Creates a new copy of the current instance, cloning its attributes into a new instance.
+     *
+     * @return A new copy of {@link RetryOptions}.
+     */
+    @Override
+    public Object clone() {
+        RetryOptions clone;
+        try {
+            clone = (RetryOptions) super.clone();
+        } catch (CloneNotSupportedException e) {
+            clone = new RetryOptions();
+        }
+
+        clone.delay(delay);
+        clone.maxDelay(maxDelay);
+        clone.maxRetries(maxRetries);
+        clone.tryTimeout(tryTimeout);
+        clone.retryMode(retryMode);
+
+        return clone;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (!(obj instanceof RetryOptions)) {
+            return false;
+        }
+
+        final RetryOptions other = (RetryOptions) obj;
+
+        return this.maxRetries() == other.maxRetries()
+            && this.retryMode() == other.retryMode()
+            && Objects.equals(this.maxDelay(), other.maxDelay())
+            && Objects.equals(this.delay(), other.delay())
+            && Objects.equals(this.tryTimeout(), other.tryTimeout());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
```

Comment:
It would be useful to also override `toString` method to simplify logging retry options.

Reply:
If we had a common way of implementing toString, I would, but it becomes a mash of random formats.